pennylane/tape/qscript.py

```diff
@@ -347,11 +347,15 @@ def batch_size(self):
         Returns:
             int or None: The batch size of the quantum script if present, else ``None``.
         """
+        if self._batch_size is _UNSET_BATCH_SIZE:
+            self._update_batch_size()
         return self._batch_size
 
     @property
     def output_dim(self):
         """The (inferred) output dimension of the quantum script."""
+        if self._output_dim is None:
+            self._update_output_dim()  # this will set _batch_size if it isn't already
```

Comment:
Is this true? I think we actually need to call `_update_batch_size` before `_update_output_dim`, also according to the comment in `_update` :thinking:

Reply:
Ah, of course! Thanks :)